test/no-complex-iteratee-expression.mjs

```diff
@@ -0,0 +1,240 @@
+import outdent from 'outdent';
+import {getTester} from './utils/test.mjs';
+
+const {test} = getTester(import.meta);
+
+const MESSAGE_ID = 'no-complex-iteratee-expression';
+
+test.snapshot({
+	valid: [
+		'for (const x of array) {}',
+		'for (const x of obj.prop) {}',
+		'for (const x of obj.prop.prop) {}',
+		'for (const x of obj.prop.prop.prop) {}',
+		'for (const x of obj.prop.prop.prop.method()) {}',
+		'for (const x of obj.methodWithNoArgs()) {}',
+		'for (const x of funcWithNoArgs()) {}',
+		'for (const x of Object.keys(obj)) {}',
+		'for (const x of Object.values(obj)) {}',
+		'for (const [x, y] of Object.entries(obj)) {}'
+	],
+	invalid: [
+		// Method call with arguments
+		{
+			code: 'for (const value of array.prop.prop.filter(shouldKeep)) {}',
+			errors: [{messageId: MESSAGE_ID}],
+			output: outdent`
+				const values = array.prop.prop.filter(shouldKeep);
+				for (const value of values) {}
+			`
+		},
+		{
+			code: 'for (const value of array.prop.prop.method(...someVariable)) {}',
```

Comment:
`snapshot` test don't need write output, that's why it's easier to test..

Reply:
Ohh ok, I'll look into that after I'm done with the braces 😄